### ISSUE TITLE
asl-locomotive error using vtk7.1.1

### ISSUE THREAD

**Issue by mik13ST** [AvtechScientific/ASL#32]
Hello,
I have just successfully compiled VTK7.1.1 and ASL. But when I try to run asl-locomotive as shown in the [Quick Start](http://asl.org.il/documentation/) I get this error:

> mik@mikPC:~/ASL$ asl-locomotive --input locomotive.stl --dx 1 --dt 2
> asl-locomotive: error while loading shared libraries: libvtkIOGeometry-7.1.so.1: cannot open shared object file: No such file or directory

I used VTK version 7.1.1 (downloaded from [here](http://www.vtk.org/files/release/7.1/VTK-7.1.1.tar.gz)), no errors during `cmake` nor `make -j4`.

However there were a few errors when complling ASL. I used `cmake -D VTK_DIR=../VTK-build/ -G "Unix Makefiles" ../ASL` to build ASL. 
These were the errors, since they mention `opencl.h` I do not consider them to be related to this.
[sample_errors.txt](https://github.com/AvtechScientific/ASL/files/1051866/sample_errors.txt)

I also tried other versions of VTK (libvtk6 from Ubuntu repos, libvtk6.3, compiled 7.0.0, this 7.1.1 and even 8.x.x RC ... none of them however got me this far)

My system is Ubuntu 17.04 64bit.

**Comment by AvtechScientific** [comment]
Hello,

try to install ASL (and VTK) from the Ubuntu repository:

https://packages.ubuntu.com/search?keywords=libasl-dev

**Comment by mik13ST** [comment]
I tried that too, but was not really sure how to use that package. Does this install the ASL the same way that `make install` does? I probably lack a bit of Unix knowledge and experience regarding working with libs. Is this mentioned in `asl-doc` package? How can I browse the documentation from that package?

Thank you for teaching me basics :+1:

EDIT: I found out how to use the documentation, but it's probably no use to me at this point.

**Comment by AvtechScientific** [comment]
You'll be able to use the package as described in the Quick Start.
Here is the list of files and their locations after the install:
https://packages.ubuntu.com/zesty/amd64/libasl-dev/filelist

and for 'asl-doc' in html format:
https://packages.ubuntu.com/zesty/all/asl-doc/filelist

**Comment by mik13ST** [comment]
I think I have progressed a bit, but still the same error.
I copied the CMakeLists.txt from Quick Start into separate dir (ASL version had to be edited from 0.1.4 to 0.1.7), placed locomotive.cc next to it, then I ran `cmake ..` from `b` sub dir. Then I ran `make`. A file named `locomotive` appeared, but when I run it using the same input args it fails the exactly same way (not finding the `libvtkIOGeometry-7.1.so.1` again).

I have the `libasl` package installed (as well as dependent `libvtk6.3`).

This is how my directory looks like:
[ASLtest.zip](https://github.com/AvtechScientific/ASL/files/1053212/ASLtest.zip)

Thanks for help so far.

EDIT: Should I try to `make install` a built VTK7.x? I always linked it using "VTK_DIR" while running `cmake` of ASL.

**Comment by AvtechScientific** [comment]
I don't know with regards to Ubuntu, but some packages install examples by default, so you should be able to run asl-locomotive right after the install. If it's not the case - download ASL from the master branch, as this issue was fixed for VTK7.1.0 (see https://github.com/AvtechScientific/ASL/issues/25)

**Comment by AvtechScientific** [comment]
Did this solve your issue? Can we close it?

**Comment by mik13ST** [comment]
Sorry,
yes, you I will am closing it.
I managed to solve all issues during compilation and runtime of ASL (latest compiled ASL with libvtk6 package). The only issue I came across was, that the locomotive.pvsm file, that is supplied [here](http://asl.org.il/input_data/locomotive.pvsm) assumed `locomotive_in_tunnel.*` files instead of `locomotive.*` files that are created in the tutorial / Quick Start.

I am unsure if this issue should be taken seriously, it might have been platform specific or something.

I think I managed to compute the something I wanted. Now I will seek a bit of professional help regarding aerodynamics. Thanks for help.
